### PR TITLE
feat(schema): send a clear message when user wants to access to an unknown field

### PR DESCRIPTION
**Pull request by Scra3**
linked to: CU-86c0hwqkb

This pull request addresses the unclear message issue discussed in https://github.com/ForestAdmin/agent-nodejs/issues/1181. The code frequently accesses fields without verifying their existence, leading to unclear messages.

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by Scra3** [comment]
Before:
image
After:
image

**Review by Thenkei — APPROVED** [review]
- ✅&nbsp;Code review 👌 
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✅&nbsp; Integration tests
- ✅&nbsp;Manual tests - I tried with fields, relations, collections it seems 🚀 
- ✅&nbsp;PR title
- ❓&nbsp;PR linked to the clickup task - Is it linked ?

Looking great.
Screenshot 2024-10-03 at 17 34 16

File: packages/agent/src/routes/access/chart.ts [hunk 1]
```diff
@@ -238,10 +238,6 @@ export default class ChartRoute extends CollectionRoute {
         value: row.value as number,
       }));
     }
-
-    throw new ValidationError(
```
Comment (Thenkei) [inline packages/agent/src/routes/access/chart.ts:239]: Just a question, with did you remove this one ?

Comment (Scra3) [reply packages/agent/src/routes/access/chart.ts:239]: I think your comment is relevant. I thought it was unreachable code, but in the end, I think it is. I will revisit it tomorrow; I think it happens when we go through a relationship that is not many-to-many or one-to-many.

Comment (Scra3) [reply packages/agent/src/routes/access/chart.ts]: fixed

File: packages/agent/src/utils/forest-schema/generator-fields.ts [hunk 2]
```diff
@@ -111,11 +111,7 @@ export default class SchemaGeneratorFields {
 
     if (relation.type === 'OneToMany') {
       targetName = relation.originKeyTarget;
-      targetField = SchemaUtils.getColumn(
-        foreignCollection.schema,
-        targetName,
-        foreignCollection.name,
-      );
+      targetField = SchemaUtils.getColumn(collection.schema, targetName, collection.name);
```
Comment (Thenkei) [inline packages/agent/src/utils/forest-schema/generator-fields.ts:114]: 👀

Comment (Thenkei) [reply packages/agent/src/utils/forest-schema/generator-fields.ts:114]: It was on foreignCollection previously.
```suggestion
      targetField = SchemaUtils.getColumn(foreignCollection.schema, targetName, foreignCollection.name);
```

File: packages/agent/src/utils/forest-schema/generator-fields.ts [hunk 3]
```diff
@@ -178,7 +174,11 @@ export default class SchemaGeneratorFields {
       relation.originKeyTarget,
       collection.name,
     );
-    const keyField = SchemaUtils.getColumn(collection.schema, relation.originKey, collection.name);
+    const keyField = SchemaUtils.getColumn(
```
Comment (Thenkei) [inline packages/agent/src/utils/forest-schema/generator-fields.ts:177]: And this one on collection

**Review by Thenkei — APPROVED** [review]

**Pull request merged**